From 59ac9596a37e3f453311c27d6c2f2a797b892e8a Mon Sep 17 00:00:00 2001
From: Algea Cao <algea.cao@rock-chips.com>
Date: Thu, 7 Jan 2021 20:14:15 +0800
Subject: [PATCH] drm/rockchip: Set picture aspect to none when show kernel
 logo

Crtc mode will include picture aspect and generate umode's flags
will include picture aspect mask when show kernel logo. After hwc
start, DRM_CLIENT_CAP_ASPECT_RATIO will be set to 0, and the umode
hwc got will not include picture aspect. That will cause two modes
mismatch, leading to a resolution reconfiguration and cause display
err.

Signed-off-by: Algea Cao <algea.cao@rock-chips.com>
Change-Id: I4e3ad1263d9d462567d14ae4c77d7b1aa05ddb1c
---
 drivers/gpu/drm/rockchip/rockchip_drm_drv.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/gpu/drm/rockchip/rockchip_drm_drv.c b/drivers/gpu/drm/rockchip/rockchip_drm_drv.c
index 13c10e3109f6..c4a76de141d0 100644
--- a/drivers/gpu/drm/rockchip/rockchip_drm_drv.c
+++ b/drivers/gpu/drm/rockchip/rockchip_drm_drv.c
@@ -713,6 +713,7 @@ static int setup_initial_state(struct drm_device *drm_dev,
 		if (ret)
 			goto error_conn;
 
+		mode->picture_aspect_ratio = HDMI_PICTURE_ASPECT_NONE;
 		ret = drm_atomic_set_mode_for_crtc(crtc_state, mode);
 		if (ret)
 			goto error_conn;
-- 
2.35.3

